application/deed/views.py

```diff
@@ -97,3 +97,73 @@ def delete_borrower(borrower_id):
         abort(status.HTTP_404_NOT_FOUND)
     else:
         return json.dumps({'id': borrower_id}), status.HTTP_200_OK
+
+
+@deed_bp.route('/<deed_reference>', methods=['PUT'])
+def get_existing_deed_and_update(deed_reference):
+
+    # Firstly check payload coming in is valid:
+    new_deed_json = request.get_json()
+    borrowerService = BorrowerService()
+
+    error_count, error_message = validate_helper(new_deed_json)
+
+    if error_count > 0:
+        return error_message, status.HTTP_400_BAD_REQUEST
+    else:
+        # If Valid:
+        # Get Current Deed
+        result = Deed.query.filter_by(token=str(deed_reference)).first()
+        existing_deed_borrowers = result.deed['borrowers']
+        i = 0
```

Comment:
better to use something more like this http://scribu.net/blog/python-equivalents-to-phps-foreach.html